bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -458,7 +458,14 @@ pub mod pallet {
 				Self::deposit_event(Event::SyncCommitteeUpdated {
 					period: update_finalized_period,
 				});
-			};
+			} else {
+				// If there was no sync committee in the update, but the finalized header provided
+				// is in the new period, we need to move the sync committee period forward.
+				if update_finalized_period == store_period + 1 {
+					<CurrentSyncCommittee<T>>::set(<NextSyncCommittee<T>>::get());
+					<NextSyncCommittee<T>>::kill();
```

Comment:
Nit: slightly more readable

```suggestion
CurrentSyncCommittee::<T>::set(NextSyncCommittee::<T>::take());
```

Reply:
I'd prefer we not do this, instead change the logic on relayer side to ensure `update_finalized_period == store_period` for non-handover update.

Reply:
> I maybe wrong but would prefer not do this(i.e. <NextSyncCommittee<T>>::kill()), my concern here is that reset NextSyncCommittee is expensive which we should avoid unless abosolutely nessessary.

It is absolutely necessary, otherwise the light client is in an incorrect state. For example, now CurrentSyncCommittee  = period 1087 and NextSyncCommittee = period 1087, which is fundamentally incorrect and could open up weird code paths that we do not expect. It could open us up to exploits, perhaps a header that was not signed by the validator set will now be verified successfully.

Also, if we don't clear NextSyncCommittee, the next sync committee import will fail because of this check: https://github.com/Snowfork/polkadot-sdk/blob/snowbridge/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L384 This is basically what happened in prod.

> NextSyncCommittee is expensive 

This can only happen once every sync committee period, in any case, right?

> Is there a way to only change the logic on relayer side to ensure signature_period == update_finalized_period == store_period for non-handover update?

Our relayers are permissionless, so we have to do an on-chain fix for this. Otherwise anyone else could exploit this bug in our light client and send an `update for store_period + 1` and brick the bridge.

The light client allows sending an update in the next period (`update for store_period + 1`) here: https://github.com/Snowfork/polkadot-sdk/blob/snowbridge/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L404 So I think the light client spec supports this and so should we.

Reply:
Yeah, I understand the light client is in an incorrect state we need to fix with a runtime upgrade. 

What I'm trying to fix is add an [else branch](https://github.com/Snowfork/polkadot-sdk/blob/3f495e56ed01f24a29d341d8928c19cc2fd8f17e/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L376) here to check `signature_period == update_finalized_period == store_period` in  for non-handover update.

Because I can't find a code path to switch SyncCommittee for non-handover update in the spec https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#apply_light_client_update so I'm a bit hesitate to do that.

Reply:
Ok, fair enough, that's a good point.

> What I'm trying to fix is add an [else branch](https://github.com/Snowfork/polkadot-sdk/blob/3f495e56ed01f24a29d341d8928c19cc2fd8f17e/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L376) at verification phase to check signature_period == update_finalized_period == store_period for non-handover update.

What do you think the `else` here should do? Something like

```
else if update_finalized_period == store_period + 1 {
	return Error::<T>::SkippedSyncCommitteePeriod;
}
```

> Because I can't find a code path to switch current_sync_committee  for non-handover update in the spec

Me neither, but that's because the consensus spec always has a sync committee in the update, right? Imo this allows headers to be verified in the next period: https://github.com/Snowfork/polkadot-sdk/blob/3f495e56ed01f24a29d341d8928c19cc2fd8f17e/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L404 And so we should handle it.

Reply:
@yrong another practical implication if we disallow the non-sync committee rollover is that if we are in period 1086, and we've received a header in 1087, we won't be able to process any messages until we've received period 1087's sync committee update. Maybe that's OK, just mentioning it.

Reply:
> the consensus spec always has a sync committee in the update

Not sure fully understand your point. Updates withthout sync committee is definetely allowed. Search `is_sync_committee_update(update)` in the spec you can see logics accordingly. 

https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#apply_light_client_update

Reply:
>  if we are in period 1086, and we've received a header in 1087, we won't be able to process any messages until we've received period 1087's sync committee update.

Yeah, I would assume that's the expected behaviour previously before adding `syncInterimFinalizedUpdate`, right? so I'm expecting we refactor the relayer logic to still respect it.

Reply:
Ah I see! But isn't the condition to handle updates in the next period missing in the spec, then? I read through [validate_light_client_update](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#validate_light_client_update) again now and I don't see it handled.